### DISCUSSION TITLE
Using with React SSR (via Loadable Components) on Express

### DISCUSSION THREAD

**Discussion by agawley** [trpc/trpc#2513] [Q&A] [answered]
I have an app that is currently React-Redux with SSR via LoadableComponents. I migrated my API to TRPC which has been great and am increasingly realising I just don't need Redux. I've taken part of my app and removed Redux in favour of TRPC-useQuery, but I'm having trouble reasoning about how to make this work in SSR. The SSr docs assume Next.js which I don't have - I'm on Express via serverless-framework.

Was wondering if anyone has any examples of how to integrate SSR outside of Next?

Thanks!

**Comment by KATT** [comment, marked as answer]
We don't have an adapter for generic SSR outside of next.js, if you want to build one you can look at this: https://github.com/trpc/trpc/blob/main/packages/next/src/withTRPC.tsx
